pyro/contrib/autoguide/__init__.py

```diff
@@ -561,6 +562,49 @@ def _loc_scale(self, *args, **kwargs):
         return loc, scale
 
 
+class AutoTransformedNormal(AutoContinuous):
+    """
+    This implementation of :class:`AutoContinuous` uses a Transformed
+    Diagonal Normal distribution via an IAF to construct a guide
+    over the entire latent space. The guide does not depend on the model's
+    ``*args, **kwargs``.
+
+    Usage::
+
+        guide = AutoTransformedNormal(model, rank=10)
+        svi = SVI(model, guide, ...)
+
+    :param callable model: a generative model
+    :param int hidden_dim: number of hidden dimensions in the IAF
+    :param float sigmoid_bias: sigmoid bias in the IAF. Defaults to ``2.0``
+    :param str prefix: a prefix that will be prefixed to all param internal sites
+    """
+    def __init__(self, model, hidden_dim=None, sigmoid_bias=2.0, prefix="auto"):
+        self.sigmoid_bias = sigmoid_bias
+        self.hidden_dim = hidden_dim
+        super(AutoTransformedNormal, self).__init__(model, prefix)
+
+    def sample_latent(self, *args, **kwargs):
+        if self.hidden_dim is None:
+            self.hidden_dim = self.latent_dim
+        loc = pyro.param("{}_loc".format(self.prefix),
+                         lambda: torch.zeros(self.latent_dim))
+        scale = pyro.param("{}_scale".format(self.prefix),
+                           lambda: torch.ones(self.latent_dim),
+                           constraint=constraints.positive)
+        iaf = dist.InverseAutoregressiveFlow(self.latent_dim, self.hidden_dim,
+                                             sigmoid_bias=self.sigmoid_bias)
+        pyro.module("{}_iaf".format(self.prefix), iaf.module)
+        self.iaf_dist = dist.TransformedDistribution(dist.Normal(loc, scale), [iaf])
+        return pyro.sample("_{}_latent".format(self.prefix), self.iaf_dist.independent(1),
+                           infer={"is_auxiliary": True})
+
+    def _loc_scale(self, *args, **kwargs):
+        loc = pyro.param("{}_loc".format(self.prefix))
```

Comment:
if _loc_scale is used for median this doesn't make sense

Reply:
Right, you should simply omit this method